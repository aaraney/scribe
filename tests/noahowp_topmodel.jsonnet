local known_formulations = import '../known_formulations.libjsonnet';

known_formulations.noahowp_topmodel
