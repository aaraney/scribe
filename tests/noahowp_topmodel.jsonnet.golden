{
   "name": "bmi_multi",
   "params": {
      "allow_exceed_end_time": true,
      "forcing_file": "",
      "init_config": "",
      "main_output_variable": "Qout",
      "model_type_name": "NoahOWP_TOPMODEL",
      "modules": [
         {
            "name": "bmi_fortran",
            "params": {
               "allow_exceed_end_time": true,
               "forcing_file": "",
               "init_config": "NoahOWP_{{id}}.namelist",
               "library_file": "libsurfacebmi",
               "main_output_variable": "QINSUR",
               "model_type_name": "NoahOWP",
               "name": "bmi_fortran",
               "variables_names_map": {
                  "LWDN": "land_surface_radiation~incoming~longwave__energy_flux",
                  "PRCPNONC": "APCP_surface",
                  "Q2": "atmosphere_air_water~vapor__relative_saturation",
                  "SFCPRS": "land_surface_air__pressure",
                  "SFCTMP": "land_surface_air__temperature",
                  "SOLDN": "land_surface_radiation~incoming~shortwave__energy_flux",
                  "UU": "land_surface_wind__x_component_of_velocity",
                  "VV": "land_surface_wind__y_component_of_velocity"
               }
            }
         },
         {
            "name": "bmi_c",
            "params": {
               "allow_exceed_end_time": true,
               "forcing_file": "",
               "init_config": "Topmodel_{{id}}.dat",
               "library_file": "libtopmodelbmi",
               "main_output_variable": "Qout",
               "model_type_name": "TOPMODEL",
               "name": "bmi_c",
               "registration_function": "register_bmi_topmodel",
               "variables_names_map": {
                  "atmosphere_water__liquid_equivalent_precipitation_rate": "QINSUR",
                  "water_potential_evaporation_flux": "EVAPOTRANS"
               }
            }
         }
      ],
      "name": "bmi_multi"
   }
}
